src/lib.rs

```diff
@@ -58,59 +68,186 @@
+    /// Creates a new menu with given `items`. It calls [`Menu::new`] and [`Menu::append_items`] internally.
+    pub fn with_items(items: &[&dyn MenuItemExt]) -> Self {
+        let menu = Self::new();
+        menu.append_items(items);
+        menu
+    }
+
+    /// Add a menu item to the end of this menu.
+    ///
+    /// ## Platform-spcific:
+    ///
+    /// - **macOS:** Only [`Submenu`] can be added to the menu
+    pub fn append(&self, item: &dyn MenuItemExt) {
+        self.0.append(item)
+    }
+
+    /// Add menu items to the end of this menu. It calls [`Menu::append`] in a loop.
+    ///
+    /// ## Platform-spcific:
+    ///
+    /// - **macOS:** Only [`Submenu`] can be added to the menu
+    pub fn append_items(&self, items: &[&dyn MenuItemExt]) {
+        for item in items {
+            self.append(*item);
+        }
+    }
+
+    /// Add a menu item to the beginning of this menu.
+    ///
+    /// ## Platform-spcific:
+    ///
+    /// - **macOS:** Only [`Submenu`] can be added to the menu
+    pub fn prepend(&self, item: &dyn MenuItemExt) {
+        self.0.prepend(item)
+    }
+
+    /// Add menu items to the beginning of this menu.
+    /// It calls [`Menu::prepend`] on the first element and
+    /// passes the rest to [`Menu::insert_items`] with position of `1`.
+    ///
+    /// ## Platform-spcific:
+    ///
+    /// - **macOS:** Only [`Submenu`] can be added to the menu
+    pub fn prepend_items(&self, items: &[&dyn MenuItemExt]) {
+        self.prepend(items[0]);
+        self.insert_items(&items[1..], 1);
+    }
+
+    /// Insert a menu item at the specified `postion` in the menu.
+    ///
+    /// ## Platform-spcific:
+    ///
+    /// - **macOS:** Only [`Submenu`] can be added to the menu
+    pub fn insert(&self, item: &dyn MenuItemExt, position: usize) {
+        self.0.insert(item, position)
+    }
+
+    /// Insert menu items at the specified `postion` in the menu.
+    ///
+    /// ## Platform-spcific:
+    ///
+    /// - **macOS:** Only [`Submenu`] can be added to the menu
+    pub fn insert_items(&self, items: &[&dyn MenuItemExt], position: usize) {
+        for (i, item) in items.iter().enumerate() {
+            self.insert(*item, position + i)
+        }
+    }
+
+    /// Remove a menu item from this menu.
     ///
-    /// ## Platform-specific:
+    /// ## Panics
     ///
-    /// - **Windows / Linux:** The menu label can contain `&` to indicate which letter should get a generated accelerator.
-    /// For example, using `&File` for the File menu would result in the label gets an underline under the `F`,
-    /// and the `&` character is not displayed on menu label.
-    /// Then the menu can be activated by press `Alt+F`.
-    pub fn add_submenu<S: AsRef<str>>(&mut self, label: S, enabled: bool) -> Submenu {
-        Submenu(self.0.add_submenu(label, enabled))
+    /// - If `item` has already been removed
+    /// - If `item` wasn't previously [append](Menu::append)ed to this menu
```

Comment:
Wouldn't a graceful failure (such as taking no action) be better? Or maybe returning a `Result` if we want to let the caller know there was a problem, though I don't think that would be necessary here.

Reply:
I will add error handling in another PR, the library shouldn't panic at most cases.